header-rewrite-filter/utility.cc

```diff
@@ -63,6 +84,69 @@ bool requiresArgument(MatchType match_type) {
     return (match_type == MatchType::Exact || match_type == MatchType::Substr || match_type == MatchType::Prefix);
 }
 
+std::tuple<absl::Status, std::string> getHeaderValue(Http::RequestOrResponseHeaderMap& headers, std::vector<absl::string_view> arguments) {
+    const Http::LowerCaseString header_key(arguments.at(0));
+    const Envoy::Http::HeaderUtility::GetAllOfHeaderAsStringResult header = Envoy::Http::HeaderUtility::getAllOfHeaderAsString(headers, header_key);
+
+    if (header.result() == absl::nullopt) { // header does not exist
+        std::cerr << "header value was not found" << std::endl;
```

Comment:
I think this could be omitted, is it just for debugging locally? The other errors seem useful though - since your bazel deps include `@envoy//source/common/common:minimal_logger_lib` you should be able to use `ENVOY_LOG` https://github.com/envoyproxy/envoy/blob/main/source/common/common/logger.h#L552. Could you use that instead? That way we can keep the logs in and enable it dynamically from the Envoy admin page